src/point.rs

```diff
@@ -64,12 +64,22 @@ impl Point {
     }
 
     /// Euclidean distance.
+    ///
+    /// # See also
+    ///
+    /// * [`Point::distance_squared`]
```

Comment:
This is a useful cross-link. It probably should be explained in prose instead of a generic see also, but this is an improvement.